static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -189,35 +191,35 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
     ])
   }
 
+  if (
+    coupon &&
+    coupon.content_type === COUPON_CONTENT_TYPE_COURSE &&
+    coupon.object_id === course.id
+  ) {
+    messages.push({ message: makeCouponMessage(coupon) })
+  }
```

Comment:
we might want to add a test to just make sure that this message comes up - what do you think?

Reply:
I looked for it and did not find any related test, otherwise I have already updated/added. If it is necessary to add, can you guid me where this test should be added as I don't have any experience of micromasters.

Reply:
No problem, for micromasters and also open-discussions we put our JavaScript tests right next to the JavaScript files. If you are editing `foo.js`, tests for that module will be in the same directory, in a file called `foo_test.js`. 

So for this file you can see tests in `StatusMessages.js`. In there you can find a lot of tests for this `calculateMessages` function, and by reading those you can probably get an idea for how to set up the test data to write the test you need. Let me know if you have other questions!

Reply:
@aliceriot thanks for the explanation, I will add the test.